packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2177,6 +2177,20 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             ? this.summarizer.summarizeOnDemand(...args)
             : this.summaryManager.summarizeOnDemand(...args);
     };
+
+    public readonly enqueueSummarize: ISummarizer["enqueueSummarize"] = (...args) => {
+        if (this.clientDetails.type === summarizerClientType) {
+            return this.summarizer.enqueueSummarize(...args);
+        } else if (this.summaryManager !== undefined) {
+            return this.summaryManager.enqueueSummarize(...args);
+        } else {
+            // If we're not the summarizer, and we don't have a summaryManager, we expect that
+            // generateSummaries is turned off.
+            throw new Error(
+                `Can't summarize, generateSummaries: ${this.runtimeOptions.summaryOptions.generateSummaries}`,
```

Comment:
We should probably call out that can't enqueue summary because this is not the right client or something like that.

Reply:
This is just a straight copy from existing PR